server/matchmaker/search.py

```diff
@@ -233,4 +234,71 @@ def __repr__(self):
         return f"Search({[p.login for p in self.players]})"
 
 
+class CombinedSearch(Search):
+    def __init__(self, *searches: Search):
+        assert searches
+        rating_type = searches[0].rating_type
+        assert all(map(lambda s: s.rating_type == rating_type, searches))
+
+        self.rating_type = rating_type
+        self.searches = searches
+        self._failed_matching_attempts = 0
```

Comment:
Do searches remain combined between matching attempts (= queue pops)?
If not I don't think this field makes much sense.
(I can't yet tell whether a party of two signing up together is a `CombinedSearch` or whether two parties of one will be combined into a `CombinedSearch` during matchmaking.)